rclrs/src/context.rs

```diff
@@ -3,73 +3,92 @@ use crate::rcl_bindings::*;
 use crate::Node;
 use alloc::sync::Arc;
 use alloc::vec::Vec;
-use cstr_core::{c_char, CString};
+use std::string::String;
 
 #[cfg(not(feature = "std"))]
-use spin::{Mutex, MutexGuard};
+use cstr_core::{c_char, CString};
+#[cfg(not(feature = "std"))]
```

Comment:
no_std is currently broken – do we still want to spend effort on no_std? If so, I can create a CI job for it.

Reply:
I'm leaning towards no, the idea is that we'd merge efforts with `r2r`, but I think it's not worth the work and both projects are currently going in their own direction. @jhdcs are you ok with dropping support for `no_std`? Thanks for working on it, and sorry if you feel your work was wasted.

Reply:
If `no_std` is broken, I'm okay with dropping support for it.

Perhaps someday, if/when we want to see about making this work with embedded devices, we can revisit `no_std` support. But for now, I think that getting the client library working for (probably) most use cases would be the best call.

So, in a nutshell, yes. Feel free to drop `no_std` support.

Reply:
It wouldn't be too much work to restore `no_std` support, so that probably shouldn't be weighted too heavily. For me, it's more about the extra work required by contributors in the future. But in the end, I don't have an opinion on whether to keep it, since I don't have a good grasp of how in-demand this feature would be.

Anyway, good to know this issue is not a blocker for this PR.